user/pages/03.commerce2/02.developer-guide/03.core/01.conditions/docs.md

```diff
@@ -4,8 +4,284 @@ taxonomy:
+shipping method. 
+
+This is explained concisely in the [\Drupal\commerce\Annotation CommerceCondition class](https://cgit.drupalcode.org/commerce/tree/src/Annotation/CommerceCondition.php) comments.
+
+```
+class CommerceCondition extends Plugin {
+// ... 
+/**
+ * The parent entity type ID.
+ *
+ * This is the entity type ID of the entity that embeds the conditions.
+ * For example: 'commerce_promotion'.
+ *
+ * When specified, a condition will only be available on that entity type.
+ *
+ * @var string
+ */
+public $parent_entity_type;
+}
+```
+
+
+Build Configuration Form
+---
+
+A condition configuration form is built just like any other form using Form 
+API. See the screenshot above to see how the config form will be displayed.
+
+When adding a new component to a condition's configuration form, be sure to 
+use the same setting name when declaring, calling, and setting 
+defaultConfiguration. See emphasis of `***'operator'***`.
+
+```
+public function buildConfigurationForm(array $form, FormStateInterface $form_state) {
+  $form = parent::buildConfigurationForm($form, $form_state);
+
+  $form[***'operator'***] = [
+    '#type' => 'select',
+    '#title' => t('Operator'),
+    '#options' => $this->getComparisonOperators(),
+    '#default_value' => $this->configuration[***'operator'***],
+    '#required' => TRUE,
+  ];
+  
+  $form['quantity'] = [
+    '#type' => 'number',
+    '#title' => t('Quantity'),
+    '#default_value' => $this->configuration['quantity'],
+    '#min' => 1,
+    '#required' => TRUE,
+  ];
+
+  return $form;
+}
+```
+
+It is important not to hardcore values for a component's `'#default_value'` or 
+`'#value'` in the `buildConfigurationForm` function. Doing so will stop any new
+configuration values set by user input from being saved. Use 
+`defaultConfiguration` function to set default values. 
+
+
+Default Configuration Settings
+---
+
+Emphasis of setting name `***'operator'***` is to show how this name should be 
+used across all functions in this class. 
+
+```
+public function defaultConfiguration() {
+  return [
+    ***'operator'*** => '>',
+    'quantity' => 1,
+  ] + parent::defaultConfiguration();
+}
+```
+
+
+Submit Configuration Form
+---
+
+This is where user inputted values are saved to the form configuration settings.
+
+```
+public function submitConfigurationForm(array &$form, FormStateInterface $form_state) {
+  parent::submitConfigurationForm($form, $form_state);
+
+  $values = $form_state->getValue($form['#parents']);
+  
+// Note that the setting name here needs to be consistent with defaultConfig
+// and buildConfigurationForm.
+  $this->configuration[***'operator'***] = $values[***'operator'***];
+  $this->configuration['quantity'] = $values['quantity'];
+}
+```
+
+
+Evaluate Function
+---
+
+This is the guts of a condition. The goal of `evaluate` is to decide if an 
+order meets a certain criteria and to return a boolean TRUE if so. 
```

Comment:
It's redundant to say "boolean TRUE". You could say that it's to decide whether the condition applies or not. That it's a boolean is implied by that description and is explicit in the condition interface.